website_sale_product_minimal_price/models/product_template.py

```diff
@@ -0,0 +1,114 @@
+                or (
+                    i.applied_on == "2_product_category" and i.categ_id == self.categ_id
+                )
+                or (i.applied_on == "1_product" and i.product_tmpl_id == self)
+                or (
+                    i.applied_on == "0_product_variant"
+                    and i.product_id in self.product_variant_ids
+                )
+            )
+            and i.compute_price == "formula"
+            and i.base == "pricelist"
+        ).mapped("base_pricelist_id")
+
+    def _get_variants_from_pricelist(self, pricelist_ids):
+        return pricelist_ids.mapped("item_ids").filtered(
+            lambda i: i.product_id in self.product_variant_ids
+        )
+
+    def _get_pricelist_variant_items(self, pricelist_id):
+        res = self._get_variants_from_pricelist(pricelist_id)
+        next_pricelists = self._get_product_subpricelists(pricelist_id)
+        res |= self._get_variants_from_pricelist(next_pricelists)
+        visited_pricelists = pricelist_id
+        while next_pricelists:
+            pricelist = next_pricelists[0]
+            if pricelist not in visited_pricelists:
+                res |= self._get_variants_from_pricelist(pricelist)
+                next_pricelists |= self._get_product_subpricelists(pricelist)
+                next_pricelists -= pricelist
+                visited_pricelists |= pricelist
+            else:
+                next_pricelists -= pricelist
+        return res
+
+    def _get_cheapest_info(self, pricelist):
+        """Helper method for getting the variant with lowest price."""
+        # TODO: Cache this method for getting better performance
+        self.ensure_one()
+        context = dict(self.env.context, pricelist=pricelist.id)
+        min_price = 99999999
+        product_id = False
+        add_qty = 0
+        has_distinct_price = False
+        # Variants with extra price
+        variants_extra_price = self.product_variant_ids.filtered("price_extra")
+        variants_without_extra_price = self.product_variant_ids - variants_extra_price
+        # Avoid compute prices when pricelist has not item variants defined
+        variant_items = self._get_pricelist_variant_items(pricelist)
+        if variant_items:
+            # Take into account only the variants defined in pricelist and one
+            # variant not defined to compute prices defined at template or
+            # category level. Maybe there is any definition on template that
+            # has cheaper price.
+            variants = variant_items.mapped("product_id")
+            products = variants + (self.product_variant_ids - variants)[:1]
+        else:
+            products = variants_without_extra_price[:1]
+        products |= variants_extra_price
+        for product in products:
+            for qty in [1, 99999999]:
+                context = dict(context, quantity=qty)
+                product_price = product.with_context(**context).price
```

Comment:
```suggestion
                product_price = product.with_context(quantity=qty).price
```

Reply:
> 

But in this context pricelist was previously included, wouldn't that parameter be necessary as well?
product_price = product.with_context(pricelist=pricelist.id, quantity=qty).price

Reply:
That object already includes de context so it's redundant

Reply:
Where does it acquire this context? In any case it should be redundant because if I remove the pricelist from this context then it does not add the text "From" to the products that have some extra price.